packages/wallet-ui/src/components/ui/organism/Header/SendModal/SendModal.view.tsx

```diff
@@ -64,7 +69,47 @@ export const SendModalView = ({ closeModal }: Props) => {
         break;
       case 'address':
         if (fieldValue !== '') {
-          if (!isValidAddress(fieldValue)) {
+          if (isValidAddress(fieldValue)) {
+            setResolvedAddress(fieldValue);
+            break;
+          } else if (isValidStarkName(fieldValue)) {
+            // Cancel previous request
+            controllerRef.current?.abort();
+
+            // Create a new AbortController for this request
+            const abortController = new AbortController();
+            controllerRef.current = abortController;
+
+            getAddrFromStarkNameQuery(fieldValue, {
+              signal: abortController.signal,
+            })
+              .then((address) => {
+                if (abortController.signal.aborted) return;
+                if (isValidAddress(address)) {
+                  setResolvedAddress(address);
+                } else {
+                  setResolvedAddress('');
+                  setErrors((prevErrors) => ({
+                    ...prevErrors,
+                    address: '.stark name doesn’t exist',
+                  }));
+                }
```

Comment:
as snap ensured the result, we can make it simple
```suggestion
                setResolvedAddress(address);
```